### DISCUSSION TITLE
Request and Resource binding

### DISCUSSION THREAD

**Discussion by reindert-vetter** [reindert-vetter/api-version-control#8] [General]
Is the documentation not completely clear? Then ask your question here.

**Comment by sash** [comment]
Hi! Great module. I Just tried the example out with binding a JsonResource (using Laravel 8) and got an error - `Unresolvable dependency resolving [Parameter #0 [ <required> $resource ]]` - reason being is that the resource needs a constructor parameter and therefore cannot be instantiated by the container. Is there a workaround for that?

P.S. 

I tried to see what would make it work as described in the specs. I came up with something like this in the Bind's handle method:
```php

use Illuminate\Http\Resources\Json\JsonResource;

//....

    public function handle(Request $request, Closure $next)
    {
        if (is_subclass_of($this->concrete, JsonResource::class)){
            app()->bind($this->abstract, function(Container $container){
                return new $this->concrete(new \stdClass);
            });
        } else {
            app()->bind($this->abstract, $this->concrete);
        }

        return $next($request);
    }
```

P.P.S.

I created a merge request with that: https://github.com/reindert-vetter/api-version-control/pull/17

**Reply by reindert-vetter** [reply]
Thank you! I think this will solve the problem well. Merged.
